### ISSUE TITLE
Version 1.60 failing to export some root WMOs

### ISSUE THREAD

**Issue by Atraxian** [Kruithne/wow.export#421]
Version 1.60 is failing to download some of the root WMOS from the online CASC looking for a file with id = 0 instead of the id specified by listfile.
The rest of the WMO components (groups and LOD files) are exported correctly.

Also it seems exporting map tiles no longer exports the WMOs (probably a related issue)

For example:
[09:36:37] Loading remote CASC file 0 (undefined)
[09:36:37] Failed to export world/wmo/zuldazar/forsaken/8fk_forsaken_shiplarge01.wmo (fileDataID does not exist in root: 0)
[09:36:37] Finished export (0 succeeded, 1 failed)

**Comment by Kruithne** [comment]
I'm unable to reproduce the issue on this end.

```
Starting export of 1 model items
Loading remote CASC file 2324175 (world/wmo/zuldazar/forsaken/8fk_forsaken_shiplarge01.wmo)
Loaded CASC file 2324175 from cache
Exporting WMO model 8fk_forsaken_shiplarge01 as OBJ: C:\Users\kruit\wow.export\world\wmo\zuldazar\forsaken\8fk_forsaken_shiplarge01.obj
Loading remote CASC file 515940 (dungeons/textures/testing/grey_05.blp)
Loaded CASC file 515940 from cache
Exporting WMO texture 515940 -> C:\Users\kruit\wow.export\dungeons\textures\testing\grey_05.png
Loading remote CASC file 2341330 (dungeons/textures/8fk_forsaken/8fk_shipplanks01.blp)
Loaded CASC file 2341330 from cache
Exporting WMO texture 2341330 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_shipplanks01.png
Loading remote CASC file 2434492 (dungeons/textures/8fk_forsaken/8fk_shipplanks01_blend.blp)
Loaded CASC file 2434492 from cache
Exporting WMO texture 2434492 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_shipplanks01_blend.png
Loading remote CASC file 2341332 (dungeons/textures/8fk_forsaken/8fk_metaltrim01.blp)
Loaded CASC file 2341332 from cache
Exporting WMO texture 2341332 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_metaltrim01.png
Loading remote CASC file 2341332 (dungeons/textures/8fk_forsaken/8fk_metaltrim01.blp)
Loaded CASC file 2341332 from cache
Exporting WMO texture 2341332 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_metaltrim01.png
Loading remote CASC file 312518 (dungeons/textures/effects/prettycolors2.blp)
Loaded CASC file 312518 from cache
Exporting WMO texture 312518 -> C:\Users\kruit\wow.export\dungeons\textures\effects\prettycolors2.png
Loading remote CASC file 2434493 (dungeons/textures/8fk_forsaken/8fk_metaltrim02.blp)
Loaded CASC file 2434493 from cache
Exporting WMO texture 2434493 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_metaltrim02.png
Loading remote CASC file 312518 (dungeons/textures/effects/prettycolors2.blp)
Loaded CASC file 312518 from cache
Exporting WMO texture 312518 -> C:\Users\kruit\wow.export\dungeons\textures\effects\prettycolors2.png
Loading remote CASC file 2434494 (dungeons/textures/8fk_forsaken/8fk_windows01.blp)
Loaded CASC file 2434494 from cache
Exporting WMO texture 2434494 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_windows01.png
Loading remote CASC file 312518 (dungeons/textures/effects/prettycolors2.blp)
Loaded CASC file 312518 from cache
Exporting WMO texture 312518 -> C:\Users\kruit\wow.export\dungeons\textures\effects\prettycolors2.png
Loading remote CASC file 2434494 (dungeons/textures/8fk_forsaken/8fk_windows01.blp)
Loaded CASC file 2434494 from cache
Exporting WMO texture 2434494 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_windows01.png
Loading remote CASC file 365609 (dungeons/textures/effects/icecolors.blp)
Loaded CASC file 365609 from cache
Exporting WMO texture 365609 -> C:\Users\kruit\wow.export\dungeons\textures\effects\icecolors.png
Loading remote CASC file 2434495 (dungeons/textures/8fk_forsaken/8fk_windows01_emv.blp)
Loaded CASC file 2434495 from cache
Exporting WMO texture 2434495 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_windows01_emv.png
Loading remote CASC file 2434496 (dungeons/textures/8fk_forsaken/8fk_metaltrim03.blp)
Loaded CASC file 2434496 from cache
Exporting WMO texture 2434496 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_metaltrim03.png
Loading remote CASC file 903711 (dungeons/textures/testing/grey_07.blp)
Loaded CASC file 903711 from cache
Exporting WMO texture 903711 -> C:\Users\kruit\wow.export\dungeons\textures\testing\grey_07.png
Loading remote CASC file 312518 (dungeons/textures/effects/prettycolors2.blp)
Loaded CASC file 312518 from cache
Exporting WMO texture 312518 -> C:\Users\kruit\wow.export\dungeons\textures\effects\prettycolors2.png
Loading remote CASC file 2434497 (dungeons/textures/8fk_forsaken/8fk_metaltrim04.blp)
Loaded CASC file 2434497 from cache
Exporting WMO texture 2434497 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_metaltrim04.png
Loading remote CASC file 312518 (dungeons/textures/effects/prettycolors2.blp)
Loaded CASC file 312518 from cache
Exporting WMO texture 312518 -> C:\Users\kruit\wow.export\dungeons\textures\effects\prettycolors2.png
Loading remote CASC file 2434498 (dungeons/textures/8fk_forsaken/8fk_skull01.blp)
Loaded CASC file 2434498 from cache
Exporting WMO texture 2434498 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_skull01.png
Loading remote CASC file 312518 (dungeons/textures/effects/prettycolors2.blp)
Loaded CASC file 312518 from cache
Exporting WMO texture 312518 -> C:\Users\kruit\wow.export\dungeons\textures\effects\prettycolors2.png
Loading remote CASC file 2434499 (dungeons/textures/8fk_forsaken/8fk_woodpost01.blp)
Loaded CASC file 2434499 from cache
Exporting WMO texture 2434499 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_woodpost01.png
Loading remote CASC file 2434500 (dungeons/textures/8fk_forsaken/8fk_woodpost02.blp)
Loaded CASC file 2434500 from cache
Exporting WMO texture 2434500 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_woodpost02.png
Loading remote CASC file 2434501 (dungeons/textures/8fk_forsaken/8fk_metaltrim05.blp)
Loaded CASC file 2434501 from cache
Exporting WMO texture 2434501 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_metaltrim05.png
Loading remote CASC file 903711 (dungeons/textures/testing/grey_07.blp)
Loaded CASC file 903711 from cache
Exporting WMO texture 903711 -> C:\Users\kruit\wow.export\dungeons\textures\testing\grey_07.png
Loading remote CASC file 312518 (dungeons/textures/effects/prettycolors2.blp)
Loaded CASC file 312518 from cache
Exporting WMO texture 312518 -> C:\Users\kruit\wow.export\dungeons\textures\effects\prettycolors2.png
Loading remote CASC file 2434502 (dungeons/textures/8fk_forsaken/8fk_rooftile01.blp)
Loaded CASC file 2434502 from cache
Exporting WMO texture 2434502 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_rooftile01.png
Loading remote CASC file 903711 (dungeons/textures/testing/grey_07.blp)
Loaded CASC file 903711 from cache
Exporting WMO texture 903711 -> C:\Users\kruit\wow.export\dungeons\textures\testing\grey_07.png
Loading remote CASC file 2434499 (dungeons/textures/8fk_forsaken/8fk_woodpost01.blp)
Loaded CASC file 2434499 from cache
Exporting WMO texture 2434499 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_woodpost01.png
Loading remote CASC file 903711 (dungeons/textures/testing/grey_07.blp)
Loaded CASC file 903711 from cache
Exporting WMO texture 903711 -> C:\Users\kruit\wow.export\dungeons\textures\testing\grey_07.png
Loading remote CASC file 2434503 (dungeons/textures/8fk_forsaken/8fk_shipplanks02.blp)
Loaded CASC file 2434503 from cache
Exporting WMO texture 2434503 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_shipplanks02.png
Loading remote CASC file 2434504 (dungeons/textures/8fk_forsaken/8fk_shipplanks02_blend.blp)
Loaded CASC file 2434504 from cache
Exporting WMO texture 2434504 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_shipplanks02_blend.png
Loading remote CASC file 2434505 (dungeons/textures/8fk_forsaken/8fk_woodboard01.blp)
Loaded CASC file 2434505 from cache
Exporting WMO texture 2434505 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_woodboard01.png
Loading remote CASC file 2434506 (dungeons/textures/8fk_forsaken/8fk_woodboard02.blp)
Loaded CASC file 2434506 from cache
Exporting WMO texture 2434506 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_woodboard02.png
Loading remote CASC file 2434507 (dungeons/textures/8fk_forsaken/8fk_stairs01.blp)
Loaded CASC file 2434507 from cache
Exporting WMO texture 2434507 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_stairs01.png
Loading remote CASC file 903711 (dungeons/textures/testing/grey_07.blp)
Loaded CASC file 903711 from cache
Exporting WMO texture 903711 -> C:\Users\kruit\wow.export\dungeons\textures\testing\grey_07.png
Loading remote CASC file 2434508 (dungeons/textures/8fk_forsaken/8fk_ship_mast01.blp)
Loaded CASC file 2434508 from cache
Exporting WMO texture 2434508 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_ship_mast01.png
Loading remote CASC file 903711 (dungeons/textures/testing/grey_07.blp)
Loaded CASC file 903711 from cache
Exporting WMO texture 903711 -> C:\Users\kruit\wow.export\dungeons\textures\testing\grey_07.png
Loading remote CASC file 2434509 (dungeons/textures/8fk_forsaken/8fk_metalgrate01.blp)
Loaded CASC file 2434509 from cache
Exporting WMO texture 2434509 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_metalgrate01.png
Loading remote CASC file 2434496 (dungeons/textures/8fk_forsaken/8fk_metaltrim03.blp)
Loaded CASC file 2434496 from cache
Exporting WMO texture 2434496 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_metaltrim03.png
Loading remote CASC file 2434501 (dungeons/textures/8fk_forsaken/8fk_metaltrim05.blp)
Loaded CASC file 2434501 from cache
Exporting WMO texture 2434501 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_metaltrim05.png
Loading remote CASC file 312518 (dungeons/textures/effects/prettycolors2.blp)
Loaded CASC file 312518 from cache
Exporting WMO texture 312518 -> C:\Users\kruit\wow.export\dungeons\textures\effects\prettycolors2.png
Loading remote CASC file 2434510 (dungeons/textures/8fk_forsaken/8fk_rope01.blp)
Loaded CASC file 2434510 from cache
Exporting WMO texture 2434510 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_rope01.png
Loading remote CASC file 903711 (dungeons/textures/testing/grey_07.blp)
Loaded CASC file 903711 from cache
Exporting WMO texture 903711 -> C:\Users\kruit\wow.export\dungeons\textures\testing\grey_07.png
Loading remote CASC file 2434496 (dungeons/textures/8fk_forsaken/8fk_metaltrim03.blp)
Loaded CASC file 2434496 from cache
Exporting WMO texture 2434496 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_metaltrim03.png
Loading remote CASC file 2434494 (dungeons/textures/8fk_forsaken/8fk_windows01.blp)
Loaded CASC file 2434494 from cache
Exporting WMO texture 2434494 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_windows01.png
Loading remote CASC file 312518 (dungeons/textures/effects/prettycolors2.blp)
Loaded CASC file 312518 from cache
Exporting WMO texture 312518 -> C:\Users\kruit\wow.export\dungeons\textures\effects\prettycolors2.png
Loading remote CASC file 2434499 (dungeons/textures/8fk_forsaken/8fk_woodpost01.blp)
Loaded CASC file 2434499 from cache
Exporting WMO texture 2434499 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_woodpost01.png
Loading remote CASC file 2434511 (dungeons/textures/8fk_forsaken/8fk_woodpost_cut.blp)
Loaded CASC file 2434511 from cache
Exporting WMO texture 2434511 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_woodpost_cut.png
Loading remote CASC file 2434512 (dungeons/textures/8fk_forsaken/8fk_metaltrim06.blp)
Loaded CASC file 2434512 from cache
Exporting WMO texture 2434512 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_metaltrim06.png
Loading remote CASC file 903711 (dungeons/textures/testing/grey_07.blp)
Loaded CASC file 903711 from cache
Exporting WMO texture 903711 -> C:\Users\kruit\wow.export\dungeons\textures\testing\grey_07.png
Loading remote CASC file 312518 (dungeons/textures/effects/prettycolors2.blp)
Loaded CASC file 312518 from cache
Exporting WMO texture 312518 -> C:\Users\kruit\wow.export\dungeons\textures\effects\prettycolors2.png
Loading remote CASC file 2434513 (dungeons/textures/8fk_forsaken/8fk_doorframe01.blp)
Loaded CASC file 2434513 from cache
Exporting WMO texture 2434513 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_doorframe01.png
Loading remote CASC file 2434494 (dungeons/textures/8fk_forsaken/8fk_windows01.blp)
Loaded CASC file 2434494 from cache
Exporting WMO texture 2434494 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_windows01.png
Loading remote CASC file 312518 (dungeons/textures/effects/prettycolors2.blp)
Loaded CASC file 312518 from cache
Exporting WMO texture 312518 -> C:\Users\kruit\wow.export\dungeons\textures\effects\prettycolors2.png
Loading remote CASC file 2434514 (dungeons/textures/8fk_forsaken/8fk_windows02.blp)
Loaded CASC file 2434514 from cache
Exporting WMO texture 2434514 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_windows02.png
Loading remote CASC file 365609 (dungeons/textures/effects/icecolors.blp)
Loaded CASC file 365609 from cache
Exporting WMO texture 365609 -> C:\Users\kruit\wow.export\dungeons\textures\effects\icecolors.png
Loading remote CASC file 2434515 (dungeons/textures/8fk_forsaken/8fk_windows02_emv.blp)
Loaded CASC file 2434515 from cache
Exporting WMO texture 2434515 -> C:\Users\kruit\wow.export\dungeons\textures\8fk_forsaken\8fk_windows02_emv.png
Loading remote CASC file 2341334 (world/wmo/zuldazar/forsaken/8fk_forsaken_shiplarge01_000_lod1.blp)
Loaded CASC file 2341334 from cache
Exporting WMO texture 2341334 -> C:\Users\kruit\wow.export\world\wmo\zuldazar\forsaken\8fk_forsaken_shiplarge01_000_lod1.png
Loading remote CASC file 2434516 (world/wmo/zuldazar/forsaken/8fk_forsaken_shiplarge01_001_lod1.blp)
Loaded CASC file 2434516 from cache
Exporting WMO texture 2434516 -> C:\Users\kruit\wow.export\world\wmo\zuldazar\forsaken\8fk_forsaken_shiplarge01_001_lod1.png
Loading remote CASC file 2621635 (world/wmo/zuldazar/forsaken/8fk_forsaken_shiplarge01_002_lod1.blp)
Loaded CASC file 2621635 from cache
Exporting WMO texture 2621635 -> C:\Users\kruit\wow.export\world\wmo\zuldazar\forsaken\8fk_forsaken_shiplarge01_002_lod1.png
Loading remote CASC file 2434517 (world/wmo/zuldazar/forsaken/8fk_forsaken_shiplarge01_003_lod1.blp)
Loaded CASC file 2434517 from cache
Exporting WMO texture 2434517 -> C:\Users\kruit\wow.export\world\wmo\zuldazar\forsaken\8fk_forsaken_shiplarge01_003_lod1.png
Loading remote CASC file 2434518 (world/wmo/zuldazar/forsaken/8fk_forsaken_shiplarge01_004_lod1.blp)
Loaded CASC file 2434518 from cache
Exporting WMO texture 2434518 -> C:\Users\kruit\wow.export\world\wmo\zuldazar\forsaken\8fk_forsaken_shiplarge01_004_lod1.png
Loading remote CASC file 2341335 (world/wmo/zuldazar/forsaken/8fk_forsaken_shiplarge01_000_lod2.blp)
Loaded CASC file 2341335 from cache
Exporting WMO texture 2341335 -> C:\Users\kruit\wow.export\world\wmo\zuldazar\forsaken\8fk_forsaken_shiplarge01_000_lod2.png
Loading remote CASC file 2434520 (world/wmo/zuldazar/forsaken/8fk_forsaken_shiplarge01_001_lod2.blp)
Loaded CASC file 2434520 from cache
Exporting WMO texture 2434520 -> C:\Users\kruit\wow.export\world\wmo\zuldazar\forsaken\8fk_forsaken_shiplarge01_001_lod2.png
Loading remote CASC file 2621636 (world/wmo/zuldazar/forsaken/8fk_forsaken_shiplarge01_002_lod2.blp)
Loaded CASC file 2621636 from cache
Exporting WMO texture 2621636 -> C:\Users\kruit\wow.export\world\wmo\zuldazar\forsaken\8fk_forsaken_shiplarge01_002_lod2.png
Loading remote CASC file 2434521 (world/wmo/zuldazar/forsaken/8fk_forsaken_shiplarge01_003_lod2.blp)
Loaded CASC file 2434521 from cache
Exporting WMO texture 2434521 -> C:\Users\kruit\wow.export\world\wmo\zuldazar\forsaken\8fk_forsaken_shiplarge01_003_lod2.png
Loading remote CASC file 2434522 (world/wmo/zuldazar/forsaken/8fk_forsaken_shiplarge01_004_lod2.blp)
Loaded CASC file 2434522 from cache
Exporting WMO texture 2434522 -> C:\Users\kruit\wow.export\world\wmo\zuldazar\forsaken\8fk_forsaken_shiplarge01_004_lod2.png
Loading remote CASC file 3569602 (world/wmo/zuldazar/forsaken/8fk_forsaken_shiplarge01_000_lod3.blp)
Loaded CASC file 3569602 from cache
Exporting WMO texture 3569602 -> C:\Users\kruit\wow.export\world\wmo\zuldazar\forsaken\8fk_forsaken_shiplarge01_000_lod3.png
Loading remote CASC file 3569603 (world/wmo/zuldazar/forsaken/8fk_forsaken_shiplarge01_001_lod3.blp)
Loaded CASC file 3569603 from cache
Exporting WMO texture 3569603 -> C:\Users\kruit\wow.export\world\wmo\zuldazar\forsaken\8fk_forsaken_shiplarge01_001_lod3.png
Loading remote CASC file 3569604 (world/wmo/zuldazar/forsaken/8fk_forsaken_shiplarge01_002_lod3.blp)
Loaded CASC file 3569604 from cache
Exporting WMO texture 3569604 -> C:\Users\kruit\wow.export\world\wmo\zuldazar\forsaken\8fk_forsaken_shiplarge01_002_lod3.png
Loading remote CASC file 3569605 (world/wmo/zuldazar/forsaken/8fk_forsaken_shiplarge01_003_lod3.blp)
Loaded CASC file 3569605 from cache
Exporting WMO texture 3569605 -> C:\Users\kruit\wow.export\world\wmo\zuldazar\forsaken\8fk_forsaken_shiplarge01_003_lod3.png
Loading remote CASC file 3569606 (world/wmo/zuldazar/forsaken/8fk_forsaken_shiplarge01_004_lod3.blp)
Loaded CASC file 3569606 from cache
Exporting WMO texture 3569606 -> C:\Users\kruit\wow.export\world\wmo\zuldazar\forsaken\8fk_forsaken_shiplarge01_004_lod3.png
Loading remote CASC file 2324179 (world/wmo/zuldazar/forsaken/8fk_forsaken_shiplarge01_000.wmo)
Loaded CASC file 2324179 from cache
Loading remote CASC file 2324180 (world/wmo/zuldazar/forsaken/8fk_forsaken_shiplarge01_001.wmo)
Loaded CASC file 2324180 from cache
Loading remote CASC file 2324181 (world/wmo/zuldazar/forsaken/8fk_forsaken_shiplarge01_002.wmo)
Loaded CASC file 2324181 from cache
Loading remote CASC file 2324182 (world/wmo/zuldazar/forsaken/8fk_forsaken_shiplarge01_003.wmo)
Loaded CASC file 2324182 from cache
Loading remote CASC file 2324183 (world/wmo/zuldazar/forsaken/8fk_forsaken_shiplarge01_004.wmo)
Loaded CASC file 2324183 from cache
Loading remote CASC file 2324184 (world/wmo/zuldazar/forsaken/8fk_forsaken_shiplarge01_005.wmo)
Loaded CASC file 2324184 from cache
Successfully exported world/wmo/zuldazar/forsaken/8fk_forsaken_shiplarge01.wmo
Finished export (1 succeeded, 0 failed)
```

Based on the log you've posted, it appears that your listfile is either incomplete or out-of-date (however given the age of the model, it's most likely incomplete). Can you post your full runtime log?

**Comment by Atraxian** [comment]
I put to 0 the listfile update time to force a listfile download on restart (just to be sure), activated all exports on the top right part (texture, alpha, .skin ecc...) and tried to exporte the raw once again.
[runtime.log](https://github.com/user-attachments/files/16247131/runtime.log)
(Full log in the attachment)

Another thing I noticed is that if I remove the export of WMO groups from the options on the top right, the WMO is exported correctly.

**Comment by Marlamin** [comment]
This is fixed in [6cf1326](https://github.com/Kruithne/wow.export/commit/6cf1326c5715c5cc604f56133bc80f67cb79f2b4) and will go out with 0.1.61.
